crates/services/txpool_v2/src/pool_worker.rs

```diff
@@ -391,6 +458,12 @@ where
                 }
             }
         }
+        // Allow in tests to use a ttl of 0 for pending pool and directly
+        // trigger the removal of expired transactions
+        #[cfg(any(feature = "test-helpers", test))]
+        {
+            self.remove_executed_transactions(vec![]);
+        }
```

Comment:
What does it change, if we don't do that?

Reply:
It means that in test like https://github.com/FuelLabs/fuel-core/blob/216c2b5e3056084d4a5206d6a07e26c0850c8609/crates/services/txpool_v2/src/tests/tests_p2p.rs#L382

Reply:
But I think it still should happen after TTL passed in next Tokio task, no?

Reply:
Okay, I see that we don't have any trigger for TTL and because of that no-one will execute it. We only clean up pending pool if we have known transaction.

Maybe, we want to try to clean up TxPool each time when we insert something or have a timer that will clean it up. Because if someone only spams us with pending transactions, then the pending pool will grow.

But maybe it is fine, idk, since it is very hard to exploit in the real network. Plus, we have limit on the size of the pending pool.

Reply:
i think its fine it's each time we insert a new pending that we are checking and not checking when we insert a normal tx so I think it's ok it cannot grow more than 3 seconds if there is a new block or a new pending coming